synapse/lib/storm.py

```diff
@@ -3923,6 +3923,8 @@ async def diffgenr():
         async with await runt.snap.view.parent.snap(user=runt.user.iden) as snap:
             snap.strict = False
 
+            snap.on('warn', runt.snap.dist)
```

Comment:
`snap.onWith()` can be used as a context manager to take care of the `off()` behavior.

Reply:
I didn't add this since it should exist for the life of the parent snap, and `off()` gets called on fini.